pulsar-client/src/main/java/org/apache/pulsar/client/impl/MessageIdImpl.java

```diff
@@ -45,6 +45,18 @@ private MessageIdImpl() {
         this(-1, -1, -1);
     }
 
+    // Private constructor used only for json deserialization
+    private MessageIdImpl(String position) {
```

Comment:
should we check any invalid value other than "earliest" and "latest"?

Reply:
Yes. I think we can throw an exception in the constructor to tell the user the position is invalid.